Makefile

```diff
@@ -26,127 +26,123 @@ UNAME_OS:=$(shell uname -s)
 UNAME_ARCH:=$(shell uname -m)
 
 # PATH/Bin
-DEPENDENCIES:=.cache/dependencies/$(UNAME_OS)/$(UNAME_ARCH)
-export GOBIN?=$(abspath .cache/dependencies/bin)
-export PATH:=$(GOBIN):$(PATH)
+DEPENDENCIES:=.cache/dependencies
+DEPENDENCY_BIN:=$(abspath $(DEPENDENCIES)/bin)
+DEPENDENCY_VERSIONS:=$(abspath $(DEPENDENCIES)/$(UNAME_OS)/$(UNAME_ARCH)/versions)
+export PATH:=$(DEPENDENCY_BIN):$(PATH)
 
 # Config
 KIND_KUBECONFIG_DIR:=.cache/e2e
 KIND_KUBECONFIG:=$(KIND_KUBECONFIG_DIR)/kubeconfig
 export KUBECONFIG?=$(abspath $(KIND_KUBECONFIG))
 export GOLANGCI_LINT_CACHE=$(abspath .cache/golangci-lint)
-API_BASE:=addons.managed.openshift.io
 export SKIP_TEARDOWN?=
+KIND_CLUSTER_NAME:="addon-operator" # name of the kind cluster for local development.
 
 # Container
 IMAGE_ORG?=quay.io/app-sre
 ADDON_OPERATOR_MANAGER_IMAGE?=$(IMAGE_ORG)/addon-operator-manager:$(VERSION)
 
-# -------
-# Compile
-# -------
+# COLORS
+GREEN  := $(shell tput -Txterm setaf 2)
+YELLOW := $(shell tput -Txterm setaf 3)
+RESET  := $(shell tput -Txterm sgr0)
 
+# ---------
+##@ General
+# ---------
+
+# Default build target - must be first!
 all: \
 	bin/linux_amd64/addon-operator-manager
 
+## Display this help.
+help:
+	@echo 'Usage:'
+	@echo '  ${YELLOW}make${RESET} ${GREEN}<target>${RESET}'
+	@awk \
+	'/^[^[:space:]]+:/ { \
+		helpMessage = match(lastLine, /^## (.*)/); \
+		if (helpMessage) { \
+			helpCommand = substr($$1, 0, index($$1, ":")-1); \
+			helpMessage = substr(lastLine, RSTART + 3, RLENGTH); \
+			printf "  ${GREEN}%-22s${RESET}%s\n", helpCommand, helpMessage; \
+		} \
+	} \
+	/^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } \
+	{ lastLine = $$0 }' $(MAKEFILE_LIST)
+
+## Prints version as used by build commands.
+version:
+	@echo $(VERSION)
+.PHONY: version
+
+## Cleans cached binaries, dependencies and container image tars.
+clean:
```

Comment:
Can we also add `delete-kind-cluster` as a dependency target?

Reply:
Adding.

I was 120% sure that `kind delete cluster` will error when the cluster does not exist - it does not :)